### PR TITLE
INT-6300 - setup integration deployment github action

### PR DESCRIPTION
**Pull request by dobregons**
## Summary

This PR adds a new github action to automate the deployment of the released version in the integration-deployments repo.

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [ ] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [ ] My changes properly paginate the target service provider's API
- [ ] My changes properly handle rate limiting of the target service provider's
      API
- [ ] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [ ] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [x] I have updated the `CHANGELOG.md` file to describe my changes
- [ ] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: .github/workflows/integration-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Get version number
+        id: get-version-number
+        uses: actions/github-script@0.9.0
+        with:
+          script: |
+            const tagName = context.payload.release.tag_name
+            const versionNumber = tagName.replace("v", "")
+            core.setOutput('versionNumber', versionNumber)
+      - name: Bump integration deployment version
+        uses: JupiterOne/integration-github-actions/create-integration-deployment@INT-6300
```
Comment (dobregons) [inline .github/workflows/integration-deployment.yml:29]: Comment for me:
Change `@INT-6300` by `@v1.0.0` as soon as the first version of the repo https://github.com/JupiterOne/integration-github-actions/pull/1 is released.

File: .github/workflows/integration-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: J1 Integration deployment
+
+on:
+  release:
+    types:
+      - published
```
Comment (VDubber) [inline .github/workflows/integration-deployment.yml:6]: This restricts to what?

File: .github/workflows/integration-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+jobs:
+  j1-integration-deployment:
```
Comment (VDubber) [inline .github/workflows/integration-deployment.yml:9]: Thought... should we look for "release" PR label to indicate that a integration-deployments PR should be created? Or does the above do that?

File: .github/workflows/integration-deployment.yml (hunk 2 above)
Comment (VDubber) [reply .github/workflows/integration-deployment.yml:6]: NPM release or github release?

Comment (dobregons) [reply .github/workflows/integration-deployment.yml:6]: This restricts to github release. So at the end of the day the flow would be:
- Dev creates PR
- Dev adds `release` label and version label (`minor`, `patch`, `major`)
- Dev merges PR after getting approvals
- Changes are available in the `main` branch
- Github action `Build and Release` is triggered. This github action creates a new tag and deploy to npm. After that it'll create a new github release.
- Then this github action will be triggered and will read the version deployed from the github release metadata and create the PR in the integration deployments.

File: .github/workflows/integration-deployment.yml (hunk 3 above)
Comment (dobregons) [reply .github/workflows/integration-deployment.yml:9]: The above does that.

**Pull request merged**